x.c

```diff
@@ -2060,8 +2060,14 @@ xdrawglyphfontspecs(const XftGlyphFontSpec *specs, Glyph base, int len, int x, i
 	}
 
 	if (base.mode & ATTR_REVERSE) {
-		bg = &dc.col[defaultbg_reverse];
-		fg = &dc.col[defaultfg_reverse];
+		if (bg == fg) {
+			bg = &dc.col[defaultfg_reverse];
+			fg = &dc.col[defaultbg_reverse];
+		} else {
+			temp = fg;
+			fg = bg;
+			bg = temp;
```

Comment:
this doesn't reintroduce this?: https://github.com/gnotclub/xst/issues/75

Reply:
i don't have a few way to verify #75 

the testcase for my problem is:
```

 [35m[45m [m[m[7m[34m lie [m[m[7m[35m /home/lie [m[m[7m[1m[35m[m[45m [m[m[35m [m[m

```

Reply:
i think what you need here to fix #75 is to remove alpha component from `bg` within this if branch for inv attr